inlang/source-code/detect-json-formatting/src/detectJsonFormatting.test.ts

```diff
@@ -1,109 +1,27 @@
-import { expect, it, describe } from "vitest"
+import { expect, it } from "vitest"
 import { detectJsonFormatting } from "./detectJsonFormatting.js"
 
-describe("defaults", () => {
-	it("should return formatting values", () => {
-		expect(detectJsonFormatting().values.endWithNewLine).toBeTruthy()
-		expect(detectJsonFormatting().values.nestedKeys).toBeFalsy()
-		expect(detectJsonFormatting().values.spacing).toBe(2)
-	})
-	it("should serialize a file according to defaults"),
-		() => {
-			const json = {
-				test: "a",
-			}
-			const serialize = detectJsonFormatting().serialize
-			const newFile = serialize(json)
-
-			expect(newFile.endsWith("\n")).toBeTruthy()
-			expect(/^{\n {2}[^ ]+.*$/m.test(newFile)).toBeTruthy()
-		}
-})
-
-describe("detect formatting values", () => {
-	it("should detect ends with new line", () => {
-		// @prettier-ignore
-		const withNewLine = `{
-	"test": "test"
-}
-`
-
-		// @prettier-ignore
-		const withoutNewLine = `{
-	"test": "test"
-}`
-		expect(detectJsonFormatting(withNewLine).values.endWithNewLine).toBeTruthy()
-		expect(detectJsonFormatting(withoutNewLine).values.endWithNewLine).toBeFalsy()
-	})
-
-	it("should detect spacing", () => {
-		// @prettier-ignore
-		const with4Spaces = `{
-    "test": "test"
-}`
-
-		// @prettier-ignore
-		const withTabs = `{
-	"test": "test"
-}`
-
-		expect(detectJsonFormatting(with4Spaces).values.spacing).toBe(4)
-		expect(detectJsonFormatting(withTabs).values.spacing).toBe("\t")
-	})
-
-	it("should correctly detect the key nesting", () => {
-		const withNesting = JSON.stringify(
-			{
-				test: {
-					test: "test",
-				},
-			},
-			undefined,
-			2
-		)
-
-		const withoutNesting = JSON.stringify(
-			{
-				"test.test": "test",
-			},
-			undefined,
-			4
-		)
-
-		expect(detectJsonFormatting(withNesting).values.nestedKeys).toBeTruthy()
-		expect(detectJsonFormatting(withoutNesting).values.nestedKeys).toBeFalsy()
-	})
+it("should detect spacing", () => {
+	// test all possible spacings
+	for (const spacing of [1, 2, 3, 4, 6, 8, "\t"]) {
+		const withSpacing = `{\n${
+			spacing === "\t" ? "\t" : " ".repeat(spacing as number)
+		}"test": "test"\n}`
+		const serialize = detectJsonFormatting(withSpacing)
+		expect(serialize(JSON.parse(withSpacing))).toBe(withSpacing)
+	}
```

Comment:
@NilsJacobsen using a generic test to go through all spacing possibilities probably saved 2-3 other tests. the test coverage is still 100%